<launch>

    <node pkg="nodelet"
          type="nodelet"
          name="rgb_to_hsv"
          args="load sb_pointcloud_processing/rgb_to_hsv nodelet_manager" output="screen">
        <remap from="~/input" to="/camera/depth_registered/points" />
    </node>

    <node pkg="nodelet"
          type="nodelet"
          name="hue_filter"
          args="load pcl/PassThrough nodelet_manager" output="screen">
        <remap from="~input" to="rgb_to_hsv/output" />
        <rosparam>
            filter_field_name: h
            filter_limit_min: 40
            filter_limit_max: 170
            filter_limit_negative: False
        </rosparam>
    </node>

    <node pkg="nodelet"
          type="nodelet"
          name="saturation_filter"
          args="load pcl/PassThrough nodelet_manager" output="screen">
        <remap from="~input" to="/hue_filter/output" />
        <rosparam>
            filter_field_name: s
            filter_limit_min: 0.1
            filter_limit_max: 0.6
            filter_limit_negative: False
        </rosparam>
    </node>

    <node pkg="nodelet"
          type="nodelet"
          name="value_filter"
          args="load pcl/PassThrough nodelet_manager" output="screen">
        <remap from="~input" to="/saturation_filter/output" />
        <rosparam>
            filter_field_name: v
            filter_limit_min: 0.1
            filter_limit_max: 1.0
            filter_limit_negative: False
        </rosparam>
    </node>

    <node pkg="nodelet"
          type="nodelet"
          name="height_filter"
          args="load pcl/PassThrough nodelet_manager" output="screen">
        <remap from="~input" to="/value_filter/output" />
        <rosparam>
            filter_field_name: z
            filter_limit_min: 0
            filter_limit_max: 0.2
            filter_limit_negative: False
            input_frame: zed_left_camera
            output_frame: zed_left_camera
        </rosparam>
    </node>

</launch>
